code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -2,19 +2,22 @@
 	name = "closet"
 	desc = "It's a basic storage unit."
 	icon = 'icons/obj/closet.dmi'
-	icon_state = "closed"
-	density = 1
+	icon_state = "generic"
+	density = TRUE
 	max_integrity = 200
 	integrity_failure = 50
 	armor = list(MELEE = 20, BULLET = 10, LASER = 10, ENERGY = 0, BOMB = 10, BIO = 0, RAD = 0, FIRE = 70, ACID = 60)
-	var/icon_closed = "closed"
-	var/icon_opened = "open"
+	var/icon_closed
+	var/icon_opened
+	var/is_wardrobe
+	var/open_door_sprite = "generic_door"
+	var/shared_open_sprite
```

Comment:
I'm not sure I understand the point of this var. Setting it to null on closets that use it doesn't seem to do anything.

Reply:
It's used during initialize to set the sprite, it does not allow for "on the fly" edits, that has to be done via icon_opened

Reply:
You're right, but on the closets which have this var, you can instead set `icon_opened` directly:
```dm
	if(shared_open_sprite)
		icon_opened = shared_open_sprite
```
vs
```dm
/obj/structure/closet/mycloset
	[...]
	icon_opened = "whatever you had for shared_icon_sprite"
```

Reply:
It's only used in the initialise proc to set it, you cannot change it on the fly once the item has been created.

Reply:
I know you cannot change it on the fly. What I'm saying is this is a pointless variable. Just set `icon_opened` directly in object declaration.

Reply:
sorry im not sure why my comments duplicated there, I wrote the second thinking the first didn't go through, one mo